openedx/core/djangoapps/content_libraries/tasks.py

```diff
@@ -84,14 +90,27 @@ def _import_block(store, user_id, source_block, dest_parent_key):
     """
     def generate_block_key(source_key, dest_parent_key):
         """
-        Deterministically generate an ID for the new block and return the key
+        Deterministically generate an ID for the new block and return the key.
+        Keys are generated such that they appear identical to a v1 library with
+        the same input block_id, library name, library organization, and parent block using derived_key
         """
-        block_id = (
-            dest_parent_key.block_id[:10] +
-            '-' +
-            hashlib.sha1(str(source_key).encode('utf-8')).hexdigest()[:10]
+        if not isinstance(source_key.lib_key, LibraryLocatorV2):
+            raise ValueError("Input must be an instance of LibraryLocatorV2")
```

Comment:
```suggestion
            raise TypeError(f"Expected source library of type LibraryLocatorV2, got {source_key.lib_key} instead.")
```

minor suggestions:
* TypeError is slightly more appropriate than ValueError.
* Including the value in the exception message always makes debugging less of a headache.